### ISSUE TITLE
training strategy

### ISSUE THREAD

**Issue by LiamZhao326** [GrandpaXun242/AdaBLDM#17]
Hello, author! I noticed that in your model training setup, you specify only one object's one category for training at a time. It seems like the dataset paths need to be manually changed for each category. Does this mean I need to continuously change the training folder and run `train.py` separately for each category to train on all the data? thx for replying!

**Comment by GrandpaXun242** [comment]
Just modify the  parameters named `gpu`  in https://github.com/GrandpaXun242/AdaBLDM/blob/main/train.py#L34 and `batchsize` in https://github.com/GrandpaXun242/AdaBLDM/blob/main/train.py#L12.

**Comment by LiamZhao326** [comment]
> Just modify the parameters named `gpu` in https://github.com/GrandpaXun242/AdaBLDM/blob/main/train.py#L34 and `batchsize` in https://github.com/GrandpaXun242/AdaBLDM/blob/main/train.py#L12.

Thank you for your response. However, my current problem is that the memory required for training is too large for a single 4090 GPU. I need to distribute the training task across multiple GPUs (4 x 4090), but I haven't been successful. Could you please teach me how to do this? Thank you!

**Comment by LiamZhao326** [comment]
> Just modify the parameters named `gpu` in https://github.com/GrandpaXun242/AdaBLDM/blob/main/train.py#L34 and `batchsize` in https://github.com/GrandpaXun242/AdaBLDM/blob/main/train.py#L12.

I set batch_size to 1 and set gpus to -1, but I'm still getting the error: CUDA out of memory. Tried to allocate 114.00 MiB (GPU 3; 11.91 GiB total capacity; 10.81 GiB already allocated; 86.88 MiB free; 11.01 GiB reserved in total by PyTorch). Each of my GPUs has nearly 12 GiB of capacity, but it's still not enough. I want to know what the correct way to distribute the load is. Thank you!
